tensorflow_lite_support/ios/task/processor/sources/TFLDetectionResult.m

```diff
@@ -16,13 +16,26 @@
 
 @implementation TFLDetection
 
-@synthesize boundingBox;
-@synthesize categories;
+- (instancetype)initWithBoundingBox:(CGRect)boundingBox
+                         categories:(NSArray<TFLCategory *> *)categories {
+  self = [super init];
+  if (self) {
+    _boundingBox = boundingBox;
+    _categories = [categories copy];
```

Comment:
@khanhlvg The copy here is not a deep copy. It only prevents any changes if the passed in categories arg is an NSMutableArray. The properties of TFLCategory objects are a shallow copy. But since our TFLCategory object properties are readonly, they can't really be modified. So should we loop through and  do a copy of each element or is that unnecessary?

Reply:
No that's fine. Excessive copy will increase the latency.